engines/director/lingo/lingo-builtins.cpp

```diff
@@ -1971,27 +1970,50 @@ void LB::b_installMenu(int nargs) {
 
 	menu->setCommandsCallback(menuCommandsCallback, g_director);
 
-	debugC(3, kDebugLingoExec, "installMenu: '%s'", Common::toPrintable(menuStxt).c_str());
+	debugC(3, kDebugLingoExec, "installMenu: '%s'", Common::toPrintable(menuStxt).encode().c_str());
 
 	LingoArchive *mainArchive = g_director->getCurrentMovie()->getMainLingoArch();
 
-	for (const byte *s = (const byte *)menuStxt.c_str(); *s; s++) {
-		// Get next line
-		line.clear();
-		while (*s && *s != '\n') { // If we see a whitespace
-			if (*s == (byte)'\xc2') {
+	// Since loading the STXT converts the text to Unicode based on the platform
+	// encoding, we need to fetch the correct Unicode character for the platform.
+
+	// STXT sections use Mac-style carriage returns for line breaks.
+	// The code preprocessor converts carriage returns to line feeds.
+	const uint32 LINE_BREAK = 0x0a;
+	// Menu definitions use the character 0xc3 to denote a checkmark.
+	// For Mac, this is √. For Windows, this is Ã.
+	const uint8 CHECKMARK_CHAR = 0xc3;
+	const uint32 CHECKMARK_U32 = numToChar(CHECKMARK_CHAR);
+	const char *CHECKMARK_STR = "Ã";
```

Comment:
This is not portable. Please use hexadecimal representation

Reply:
Fixed